cmd/internal/flowmode/cmd_convert.go

```diff
@@ -0,0 +1,141 @@
+package flowmode
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/spf13/cobra"
+
+	"github.com/grafana/agent/converter"
+	convert_diag "github.com/grafana/agent/converter/diag"
+	"github.com/grafana/agent/pkg/river/diag"
+)
+
+func convertCommand() *cobra.Command {
+	f := &flowConvert{
+		output:         "",
+		sourceFormat:   "",
+		bypassWarnings: false,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "convert [flags] file",
+		Short: "Convert a supported config file to River",
+		Long: `The convert subcommand translates a supported config file to
+a River configuration file.
+
+If the file argument is not supplied or if the file argument is "-", then convert will read from stdin.
+
+The -o flag can be used to write the formatted file back to disk. -o can not be provided when convert is reading from stdin. When -o is not provided, fmt will write the result to stdout.
+
+The -f flag can be used to specify the format we are converting from.
+
+The -b flag can be used to bypass warnings.`,
+		Args:         cobra.RangeArgs(0, 1),
+		SilenceUsage: true,
+		Aliases:      []string{"convert"},
+
+		RunE: func(_ *cobra.Command, args []string) error {
+			var err error
+
+			if len(args) == 0 {
+				// Read from stdin when there are no args provided.
+				err = f.Run("-")
+			} else {
+				err = f.Run(args[0])
+			}
+
+			var diags diag.Diagnostics
+			if errors.As(err, &diags) {
+				for _, diag := range diags {
+					fmt.Fprintln(os.Stderr, diag)
+				}
+				return fmt.Errorf("encountered errors during formatting")
+			}
+
+			return err
+		},
+	}
+
+	cmd.Flags().StringVarP(&f.output, "output", "o", f.output, "The filepath where the output is written.")
+	cmd.Flags().StringVarP(&f.sourceFormat, "source-format", "f", f.sourceFormat, "The format of the source file. Supported formats: 'prometheus'.")
```

Comment:
Is there an easy way to make this a required parameter? Otherwise it fails with the cryptic message
```
Error: (Error) unrecognized kind "" given to the config converter
```